### PR TITLE
joyent/smartos-live#484 Fresh install fails to boot completely on large ram servers

### PR DESCRIPTION
**Pull request by failedrequest**
This is a small patch to the fix Fresh install fails to boot completely due to bug in code for setting dump zvol  issue #484  

In this patch we use dumpadm -e to get the estimated dumpsize for kernel pages. Since SmartOS is configured to only dump kernel pages by default this is a safe size to use.  The value from dumpadm -e is converted to MB and padded by 1mb to allow it to fix into the proper volblock . We also check to insure that the minimum size for the dumpvol is 4G . 

.

### CODE REVIEW

**Comment by nshalman** [comment]
Also usually a good idea to update the commit message to start with the bug ID being fixed. In this case it should be:
```
joyent/smartos-live#484 Fresh install fails to boot completly due to bug in code for setting dump zvol
```

If you dislike that summary you might want to edit the description of the original bug first, then set the commit to match.

**Comment by failedrequest** [comment]
Updated the code to use 2 times the estimate from dumpadm -e . It has been tested on 256G ram servers and it resolves the issue as expcted.

File: overlay/generic/smartdc/lib/smartos_prompt_config.sh [hunk 1]
```diff
@@ -751,18 +751,16 @@ promptpool()
...
-	[ ${base_size} -gt 4096 ] && base_size=4096
-
-	# Create the dump zvol
-	zfs create -V ${base_size}mb ${SYS_ZPOOL}/dump || \
-	    fatal "failed to create the dump zvol"
+	# Get the estimated dump size from dumpadm -e in mb double it and convert to int
+
+	local dumpsize=$(echo "(( $(dumpadm -e |  sed "s/[^0-9|.]//g") * 1024) * 2 )/1" |bc)
+
+	# Ensure the dumpvol is at least 4G. This will store most kernel page only dumps
```
Comment (rmustacc) [inline overlay/generic/smartdc/lib/smartos_prompt_config.sh]: Where does 4 GiB come from?

Comment (failedrequest) [inline overlay/generic/smartdc/lib/smartos_prompt_config.sh]: The 4G was changed to 1G after testing showed that 4G was a waste. The current version uses 1G as the smallest size. Which should be sufficient for a starting point for dumps.

**Comment by jclulow** [comment]
Thanks for the changes.  I will pull this into `master`.

**Comment by jclulow** [comment]
Note that in final testing I found one last issue -- the dump size estimation is formatted by `zfs_nicenum()`, which will potentially render the size in megabytes (with a suffix of `M`), or even kilobytes (`K`).  I have adjusted the estimation calculation to take this into effect, and reformatted the comment a little.

Thanks again!

**Comment by RSully** [comment]
This was done in commit 665689294042901fa6514919b70d2234a6a5eab4. This pull can probably be closed.

**Pull request closed without merging**